go/mysql/binlog_event_compression.go

```diff
@@ -98,7 +98,7 @@ func init() {
 	if err != nil { // Should only happen e.g. due to ENOMEM
 		log.Errorf("Error creating stateless decoder: %v", err)
 	}
-	statefulDecoderPool = sync.Pool{
+	statefulDecoderPool.pool = sync.Pool{
```

Comment:
When adopting this as a new struct, you don't really need the `New` option, I just put all of that logic into the `Get()` for when you have a miss from the pool. Keeps it all nicely self contained into the decoderPool.